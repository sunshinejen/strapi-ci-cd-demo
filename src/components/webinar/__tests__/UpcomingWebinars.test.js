import React from 'react';
import { getAllByRole, getAllByTestId, render, screen } from '@testing-library/react';
import UpcomingWebinars from '../upcomingWebinars';

//webinar test data 
const upcomingWebinarTestData = [
    {
        date: 'Thursday, July 09, 2037',
        upcomingWebinar_type: "webinar",
        upcoming_webinar_title: "Metaverse",
        id: "23423423423423"
    },
    {
        date: 'Thursday, July 10, 2037',
        upcomingWebinar_type: "webinar2",
        upcoming_webinar_title: "Metaverse2",
        id: "234234234893423"
    }
]
//Snapshot Test 
it("Upcoming Webinars renders as expected", () => {
    render(<UpcomingWebinars upcoming={upcomingWebinarTestData} />)
});

test("Upcoming Webinars snapshot", () => {
    const { asFragment } = render(<UpcomingWebinars upcoming={upcomingWebinarTestData} />)
    expect(asFragment()).toMatchSnapshot()
});

test("Upcoming Webinars Unit Testing", () => {
    const { getAllByRole} = render(<UpcomingWebinars upcoming={upcomingWebinarTestData} />)
    const listItems = getAllByRole("listitem");
    expect(listItems).toHaveLength(2);
})
