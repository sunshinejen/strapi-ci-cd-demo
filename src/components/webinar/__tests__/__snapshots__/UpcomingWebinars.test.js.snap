// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upcoming Webinars snapshot 1`] = `
<DocumentFragment>
  <div
    class="upcoming-Webinars"
  >
    <div
      class="container py-5"
    >
      <h5>
        Upcoming Webinars
      </h5>
      <div
        class="wrapper-grid"
      >
        <ul>
          <li
            class="webinar-items"
            data-testid="upcoming-webinar-items"
          >
            Metaverse - Thursday, July 09, 2037
          </li>
          <li
            class="webinar-items"
            data-testid="upcoming-webinar-items"
          >
            Metaverse2 - Thursday, July 10, 2037
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
